nilearn/datasets/func.py

```diff
@@ -1276,3 +1276,174 @@ def fetch_mixed_gambles(n_subjects=1, data_dir=None, url=None, resume=True,
+        resumed to download again.
+
+    verbose: int, Default is 1
+        This parameter is used to set the verbosity level to print the message
+        to give information about the processing.
+        0 indicates no information will be given.
+
+    Returns
+    -------
+    data: sklearn.datasets.base.Bunch
+        Dictionary-like object, attributes are:
+
+        'FullCorrelation': Full correlation matrices
+
+        'PartialCorrelation': Partial correlation matrices
+
+    References
+    ----------
+    For more details:
+    Stephen Smith et al, HCP beta-release of the Functional Connectivity MegaTrawl.
+    April 2015 "HCP500-MegaTrawl" release.
+
+    https://db.humanconnectome.org/megatrawl/
+
+    Disclaimer
+    ----------
+    IMPORTANT: This is open access data. You must agree to Terms and conditions
+    of using this data before using it,
+    available at: http://humanconnectome.org/data/data-use-terms/open-access.html
+    Open Access Data (all imaging data and most of the behavioral data)
+    is available to those who register an account at ConnectomeDB and agree to
+    the Open Access Data Use Terms. This includes agreement to comply with
+    institutional rules and regulations. This means you may need the approval
+    of your IRB or Ethics Committee to use the data. The released HCP data are
+    not considered de-identified, since certain combinations of HCP Restricted
+    Data (available through a separate process) might allow identification of
+    individuals. Different national, state and local laws may apply and be
+    interpreted differently, so it is important that you consult with your IRB
+    or Ethics Committee before beginning your research. If needed and upon request,
+    the HCP will provide a certificate stating that you have accepted the
+    HCP Open Access Data Use Terms. Please note that everyone who works with
+    HCP open access data must review and agree to these terms, including those
+    who are accessing shared copies of this data. If you are sharing
+    HCP Open Access data, please advise your co-researchers that they must
+    register with ConnectomeDB and agree to these terms.
+    Register and sign the Open Access Data Use Terms at
+    ConnectomeDB: https://db.humanconnectome.org/
+    """
+    url = "https://www.nitrc.org/frs/download.php/8037/Megatrawls.tgz"
+    opts = {'uncompress': True}
+
+    # dataset terms
+    dimensionalities = ['d25', 'd50', 'd100', 'd200', 'd300']
+    timeseries_methods = ['ts2', 'ts3']
+
+    message = ("The %s you have given '%s' is invalid. Please choose either "
+               "of them %s or list of specific choices.")
+
+    names = ['choice_dimensionality', 'choice_timeseries']
+    inputs = [choice_dimensionality, choice_timeseries]
+    assign_names = ['dimensionalities', 'timeseries_methods']
+    standard_inputs = [dimensionalities, timeseries_methods]
+
+    for name_, input_, assign_, standard_ in zip(names, inputs, assign_names,
```

Comment:
As a comment, two reasons I know of for using final underscores:
* avoid shadowing a builtin like list
* scikit-learn convention for attributes that depends on data

IMHO final underscores are not that great for readability, especially when you have `some_variable` and `some_variable_` in the same function.